pkg/istio-agent/agent.go

```diff
@@ -381,6 +381,14 @@ func (a *Agent) Run(ctx context.Context) (func(), error) {
 			if err := a.xdsProxy.initIstiodDialOptions(a); err != nil {
 				log.Warnf("Failed to init xds proxy dial options")
 			}
+
+			if s := a.secretCache; s != nil {
```

Comment:
Should root cert rotation be supported for all CA configurations or should it be limited to certain configurations?

Reply:
IMO, it's fine.
But maybe we can add a feature gate, wdyt?